pkg/cluster/k8sres.go

```diff
@@ -1650,7 +1776,317 @@ func (c *Cluster) generateStatefulSet(spec *cpov1.PostgresSpec) (*appsv1.Statefu
 	return statefulSet, nil
 }
 
-func generateTlsMounts(spec *cpov1.PostgresSpec, tlsEnv func(key string) string) ([]v1.EnvVar, []cpov1.AdditionalVolume) {
+func (c *Cluster) generateRepoHostStatefulSet(spec *cpov1.PostgresSpec) (*appsv1.StatefulSet, error) {
```

Comment:
Looks like there is a ton of stuff here left over from copy pasting from the db stateful set generation. Should remove all of irrelevant stuff and maybe factor out any duplication that's left over if it makes sense.